ui/app/utils/format-duration.js

```diff
@@ -0,0 +1,47 @@
+import moment from 'moment';
+
+const allUnits = [
+  { name: 'years', suffix: 'year', inMoment: true, pluralizable: true },
+  { name: 'months', suffix: 'month', inMoment: true, pluralizable: true },
+  { name: 'days', suffix: 'day', inMoment: true, pluralizable: true },
+  { name: 'hours', suffix: 'h', inMoment: true, pluralizable: false },
+  { name: 'minutes', suffix: 'm', inMoment: true, pluralizable: false },
+  { name: 'seconds', suffix: 's', inMoment: true, pluralizable: false },
+  { name: 'milliseconds', suffix: 'ms', inMoment: true, pluralizable: false },
+  { name: 'microseconds', suffix: 'µs', inMoment: false, pluralizable: false },
+  { name: 'nanoseconds', suffix: 'ns', inMoment: false, pluralizable: false },
+];
+
+export default function formatDuration(duration = 0, units = 'ns') {
+  const durationParts = {};
+
+  if (units === 'ns') {
+    durationParts.nanoseconds = duration % 1000;
+    durationParts.microseconds = Math.floor((duration % 1000000) / 1000);
+    duration = Math.floor(duration / 1000000);
+  } else if (units === 'mms') {
+    durationParts.microseconds = duration % 1000;
+    duration = Math.floor(duration / 1000);
+  }
```

Comment:
I found this part a little tricky to follow.  My reading is that the unit of `duration` can be one of nanoseconds, microseconds, or milliseconds, and for the former two cases we're normalizing it back to milliseconds here.

If that's correct, it might also be clearer to mutate the `units` argument in these blocks also, rather than in the dense ternary on the following line. And perhaps some commentary explaining this might also help?

Reply:
Units can be any unit moment supports (ms, s, m, h, days, months, years) as well as microseconds and nanoseconds, which moment doesn't support.

So this ternary:

```js
  const momentDuration = moment.duration(duration, ['ns', 'mms'].includes(units) ? 'ms' : units);
```
Uses units as is unless units is either microseconds or nanoseconds, in which case nanos and micros need to be calculated upfront and duration needs to be normalized as milliseconds.

But your suggestion to reassign `units` to `ms` is still valid. Also point taken about needing comments.